lib/subordinateio.c

```diff
@@ -812,14 +794,13 @@ int list_owner_ranges(const char *owner, enum subid_type id_type, struct subordi
 	commonio_rewind(db);
 	while ((range = commonio_next(db)) != NULL) {
 		if (0 == strcmp(range->owner, owner)) {
-			if (!append_range(&ranges, range, count++)) {
-				free_subordinate_ranges(ranges, count-1);
+			if (!append_range(ranges, range, count++)) {
+				free(*ranges);
```

Comment:
See my comment to removal of `subid_free_ranges()`: we still need to iterate over every array item first to free subordinate_range::owner...

Reply:
yeah that's ugly.  I'd forgotten about the -> owner, and, given that, would prefer to revert  '(fold up) change list_owner_ranges to do **ranges, not ***ranges'  so that the code using subordinate_range can stay consistent.

Reply:
Ok.